### PR TITLE
Upgrade WORKSPACE python packages for python3.9 support

### PR DESCRIPTION
**Pull request by Code0x58**
This upgrades a few python packages. One benefit is that infrastructure/tests work on python3.9 (due to upgrading pex to a version that says it can be installed in python <=3.9) so will fix the issue seen in #3638

### CODE REVIEW

**Review by nicknezis — APPROVED** [review]

**Comment by nicknezis** [comment]
Looks good. What else is needed to remove from draft status?

**Review by DvalCaruso — APPROVED** [review]
Duplicate of #

**Comment by joshfischer1108** [comment]
@Code0x58 What do we need to do to pick up work on this branch?  Anything I can help with?

**Comment by Code0x58** [comment]
I think with these changes the python version in Travis blows up, or image (tested via `./scripts/release/build-images --all`) having versions that blow up during the build.

I can't recall what specifically required a change from 3.6 to something higher - probably just the newer libraries that work with 3.9.

One path forward is to upgrade all the python versions installed in the container images+CI (Travis + Ubuntu vagrant manifests) to some minimum version - you could use pyenv in the Vagrant machine to find what the lowest working version is. This would be a bit of a pain due to the multiple runtimes required to use Heron, so can't just use pre-existing base images (I think the JRE may be the first class citizen for that in the images). It's not a particularly appealing change, especially considering it may force python tennants to try upgrading their code+dependencies, but it's probably a reasonable thing to do without reconsidering the whole architecture or trying stick with Python 3.6-3.8 only.

**Comment by nwangtw** [comment]
I am wondering which module uses more libraries? It might be ideal to
minimize the use of extra libraries considering the compatibility issues
with Python.

File: .travis.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ addons:
       - libgoogle-perftools-dev
 
 env:
-  - BAZEL_VERSION=4.1.0 ENABLE_HEAPCHECK=1
+  - BAZEL_VERSION=4.1.0 ENABLE_HEAPCHECK=1 LC_ALL=en_US.UTF-8 LANG=en_US.UTF-8 LANGUAGE=en_US.UTF-8
```
Comment (surahman) [inline .travis.yml:32]:
```suggestion
  - BAZEL_VERSION=4.1.0 ENABLE_HEAPCHECK=1
```

Should resolve error:

```bash
UnicodeEncodeError: 'ascii' codec can't encode characters in position 73-75: ordinal not in range(128)
```

Comment (surahman) [reply .travis.yml:32]: Reverted. The build container is still running Python 3.6:

```bash
$ which python
/opt/pyenv/shims/python
The command "which python" exited with 0.
$ python -V
Python 2.7.17
The command "python -V" exited with 0.
$ which python3
/opt/pyenv/shims/python3
The command "which python3" exited with 0.
$ python3 -V
Python 3.6.9
The command "python3 -V" exited with 0.
```

**Comment by surahman** [comment]
The build containers are still running older versions of Python:

```bash
$ which python
/opt/pyenv/shims/python
The command "which python" exited with 0.
$ python -V
Python 2.7.17
The command "python -V" exited with 0.
$ which python3
/opt/pyenv/shims/python3
The command "which python3" exited with 0.
$ python3 -V
Python 3.6.9
The command "python3 -V" exited with 0.
```
We will probably need to update the base Dockerfiles here: https://github.com/apache/incubator-heron/tree/master/docker/compile

https://github.com/apache/incubator-heron/blob/ee369ef7745df530c765e89d2e8a2eefc8d8b742/docker/compile/Dockerfile.ubuntu18.04#L23-L40

**Comment by nicknezis** [comment]
#3642 Code was integrated into this PR.

Tracker Behavior changes:

- replace `/` → `/topologies` redirect with (OpenAPI on ReDoc) documentation
- deprecate `executiontime` and always return 0.0 for it in the response envelope - this could probably be put back in, but I don't think it's worth it
- inbound requests aren't terminated if they take over 120s. At the moment, every outbound request has a 5s limit (due to httpx default) so I wouldn't expect the timeout to be needed (even in metrics queries which may transform data). A replacement timeout could be added in if necessary.

**Review by nwangtw — APPROVED** [review]

**Review by nicknezis — APPROVED** [review]

**Pull request merged**